nncf/torch/extensions/include/common_cuda_funcs.cuh

```diff
@@ -6,9 +6,12 @@
 // which is rumoured to degrade performance.
 
 #include "common_cuda_defs.cuh"
+#define DISABLE_FP16(TYPE_NAME) std::enable_if_t< \
+                     std::is_same<float, TYPE_NAME>::value || \
+                     std::is_same<double, TYPE_NAME>::value, bool> = true
```

Comment:
This macros allows to restrict possible types in functions templates. Wrapped TYPE_NAME could be only `float` or `double`, compiler doesn't build template version with FP16

Reply:
I guess you should use `at::ScalarType::Half`, i.e. use the correct namespace, then:
https://github.com/pytorch/pytorch/blob/a75edfa97c9d985a337fbea7b9c0f4061153aaf0/aten/src/ATen/Dispatch.h#L262

If it doesn't work, then you are probably just not including the right header. Search the ATen source code for the definition of these type aliases and you'll find the right header.

Reply:
`at::ScalarType::Float` and `at::ScalarType::Double` in comparison with `at::ScalarType::Half` are just elements of an enum. 
So, it's not possible to use them in template declaration